app/services/testing_ground/save_as.rb

```diff
@@ -18,14 +18,16 @@ def initialize(original, name, user = nil)
 
   def run
     duplicate = @original.dup
+    duplicate.name = @name
+    duplicate.user = @user
 
-    TestingGround.transaction do
-      duplicate.name = @name
-      duplicate.user = @user
+    if duplicate.valid?
```

Comment:
> Maybe catch prevents this behavior though.

My bad. I meant `rescue`; catch is something different altogether. Too much time writing JavaScript lately... 😆